repl/scala-2.11/src/main/scala/org/apache/spark/repl/Main.scala

```diff
@@ -43,10 +39,20 @@ object Main extends Logging {
 
   def main(args: Array[String]) {
     if (getMaster == "yarn-client") System.setProperty("SPARK_YARN_MODE", "true")
+
+    val interpArguments = List(
+      "-Yrepl-class-based",
+      "-Yrepl-outdir", s"${outputDir.getAbsolutePath}",
```

Comment:
"-usejavacp" is actually passed from the build `sparkShell` task. I didn't want to change the conditions under which it is called (i.e. it may not be needed if other arguments are supplied)